extensions/amp-story/1.0/amp-story-page-attachment.js

```diff
@@ -56,40 +68,83 @@ export class AmpStoryPageAttachment extends DraggableDrawer {
   buildCallback() {
     super.buildCallback();
 
-    this.headerEl_.appendChild(
-      htmlFor(this.element)`
-        <span class="i-amphtml-story-page-attachment-close-button"
-            role="button">
-        </span>`
-    );
-    this.headerEl_.appendChild(
-      htmlFor(this.element)`
-        <span class="i-amphtml-story-page-attachment-title"></span>`
-    );
-
-    if (this.element.hasAttribute('data-title')) {
-      this.headerEl_.querySelector(
-        '.i-amphtml-story-page-attachment-title'
-      ).textContent = this.element.getAttribute('data-title');
-    }
-
     const theme = this.element.getAttribute('theme');
     if (theme && AttachmentTheme.DARK === theme.toLowerCase()) {
       this.headerEl_.classList.add(DARK_THEME_CLASS);
       this.element.classList.add(DARK_THEME_CLASS);
     }
 
-    const templateEl = this.element.querySelector(
-      '.i-amphtml-story-draggable-drawer'
-    );
+    // URL will be validated and resolved based on the canonical URL if relative
+    // when navigating.
+    const href = this.element.getAttribute('href');
+    this.type_ = href ? AttachmentType.REMOTE : AttachmentType.INLINE;
+
+    if (this.type_ === AttachmentType.INLINE) {
```

Comment:
WDYT about making a `buildInline` and `buildRemote` function to make this more readable?

Reply:
Done